packages/next-urql/src/with-urql-client.tsx

```diff
@@ -72,27 +71,29 @@ export function withUrqlClient(
 
       return (
         <Provider value={client}>
-          <AppOrPage
-            urqlClient={client}
-            resetUrqlClient={resetUrqlClient}
-            {...rest}
-          />
+          {createElement(AppOrPage, {
+            ...rest,
+            urqlClient: client,
+            resetUrqlClient,
+          })}
         </Provider>
       );
     };
 
     // Set the displayName to indicate use of withUrqlClient.
-    withUrql.displayName = `withUrqlClient(${getDisplayName(AppOrPage)})`;
+    const displayName = AppOrPage.displayName || AppOrPage.name || 'Component';
+    withUrql.displayName = `withUrqlClient(${displayName})`;
 
     if (AppOrPage.getInitialProps || options!.ssr) {
-      withUrql.getInitialProps = async (appOrPageCtx: NextUrqlContext) => {
-        const { AppTree } = appOrPageCtx;
+      withUrql.getInitialProps = async (_ctx: any) => {
```

Comment:
Does this need to be `any` here? I believe we could just do:

```suggestion
      withUrql.getInitialProps = async (_ctx: PartialNextContext) => {
```

directly unless I'm missing something? It doesn't look like the type cast on L89 is necessary based on the way we access `appOrPageCtx`.